core/guard/guard.go

```diff
@@ -0,0 +1,117 @@
+package guard
+
+import (
+	"github.com/gogo/protobuf/proto"
+	ic "github.com/libp2p/go-libp2p-core/crypto"
+	"time"
+
+	config "github.com/TRON-US/go-btfs-config"
+	"github.com/TRON-US/go-btfs/core/commands/storage"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/tron-us/go-btfs-common/crypto"
+	guardPb "github.com/tron-us/go-btfs-common/protos/guard"
+)
+
+func NewContract(session *storage.FileContracts, configuration *config.Config, chunkHash string, chunkIndex int32) (*guardPb.ContractMeta, error) {
+	shard := session.ShardInfo[chunkHash]
+	guardPid, escrowPid, err := getGuardAndEscrowPid(configuration)
+	if err != nil {
+		return nil, err
+	}
+	return &guardPb.ContractMeta{
+		ContractId:       shard.ContractID,
+		RenterPid:        session.Renter.Pretty(),
+		HostPid:          shard.Receiver.Pretty(),
+		ShardHash:        chunkHash,
+		ShardIndex:       chunkIndex,
+		FileHash:         session.FileHash.KeyString(),
+		RentStart:        shard.StartTime,
+		RentEnd:          shard.StartTime.Add(shard.Length),
+		GuardPid:         guardPid.Pretty(),
+		EscrowPid:        escrowPid.Pretty(),
+		Price:            shard.Price,
+		Amount:           1000,
+		CollateralAmount: 0,
+		PayoutSchedule:   0,
+		NumPayouts:       5,
+	}, nil
+}
+
+func SignedContractAndMarshal(meta *guardPb.ContractMeta, cont *guardPb.Contract, privKey ic.PrivKey, isPayer bool) ([]byte, error) {
+	sig, err := crypto.Sign(privKey, meta)
+	if err != nil {
+		return nil, err
+	}
+	if cont == nil {
+		cont = &guardPb.Contract{
+			ContractMeta:   *meta,
+			LastModifyTime: time.Now(),
+		}
+	} else {
+		cont.LastModifyTime = time.Now()
+	}
+	if isPayer {
+		cont.RenterSignature = sig
+	} else {
+		cont.HostSignature = sig
+	}
+	return proto.Marshal(cont)
+}
+
+func UnmarshalGuardContract(marshaledBody []byte) (*guardPb.Contract, error) {
+	signedContract := &guardPb.Contract{}
+	err := proto.Unmarshal(marshaledBody, signedContract)
+	if err != nil {
+		return nil, err
+	}
+	return signedContract, nil
+}
+
+func getGuardAndEscrowPid(configuration *config.Config) (peer.ID, peer.ID, error) {
+	escrowPid, err := pidFromString(configuration.Services.EscrowPubKeys[0])
```

Comment:
should return err if `len(EscrowPubKeys) == 0`

Reply:
ok